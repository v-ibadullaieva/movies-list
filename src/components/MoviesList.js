import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchMovies, deleteMovie } from "../features/movies/actions";
import { Button } from "reactstrap";
import { Link } from "react-router";

export class MoviesList extends Component {
  componentDidMount() {
    this.props.fetchMovies({
      title: this.props.location.query.title || "",
      actor: this.props.location.query.actor || ""
    });
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.props.location.query.title !== nextProps.location.query.title ||
      this.props.location.query.actor !== nextProps.location.query.actor
    ) {
      this.props.fetchMovies({
        title: nextProps.location.query.title,
        actor: nextProps.location.query.actor
      });
    }
  }

  render() {
    return (
      <ul className="list">
        {this.props.movies.map(movie => (
          <li key={movie.id} className="list__item">
            <Link to={`/movies/${movie.id}`} className="list__link">
              {movie.title}
            </Link>
            <Button size="sm" onClick={() => this.props.deleteMovie(movie.id)}>
              Delete
            </Button>
          </li>
        ))}
      </ul>
    );
  }
}

export default connect(
  state => ({
    movies: Object.values(state.app.movies)
  }),
  { fetchMovies, deleteMovie }
)(MoviesList);
