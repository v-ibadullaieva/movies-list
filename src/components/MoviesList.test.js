import React from "react";
import { MoviesList } from "./MoviesList";
import renderer from "react-test-renderer";

const props = {
  movies: [
    {
      id: 99,
      title: "title",
      year: "1998",
      format: "vhs",
      actors: [{ name: "John" }]
    }
  ],
  params: { title: "", actor: "" },
  location: {
    query: {}
  },
  fetchMovies: () => {}
};

it("renders without crashing", () => {
  const tree = renderer.create(<MoviesList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
