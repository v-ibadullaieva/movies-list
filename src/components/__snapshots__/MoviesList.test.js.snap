// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="list"
>
  <li
    className="list__item"
  >
    <a
      className="list__link"
      onClick={[Function]}
      style={Object {}}
    >
      title
    </a>
    <button
      className="btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </li>
</ul>
`;
